File: acmicpc/11729_하노이 탑 이동 순서.cpp
#include <stdio.h>

int hanoi(int from, int tmp, int to, int n) {
	if (n == 0) return 0;
	hanoi(from, to, tmp, n - 1);
	printf("%d %d\n", from, to);
	hanoi(tmp, from, to, n - 1);
	return 0;
}
int main(void) {
	int n;
	int c = 1;
	scanf("%d", &n);
	for (int i = 1; i <= n; i++) c *= 2;
	printf("%d\n", c - 1);
	hanoi(1, 2, 3, n);

	return 0;
}

File: acmicpc/python/11729_하노이 탑 이동 순서.py
def hanoi(n: int, from_: int, to_: int):
    if n == 1:
        print(from_, to_)
    else:
        hanoi(n - 1, from_, 6 - from_ - to_)
        print(from_, to_)
        hanoi(n - 1, 6 - from_ - to_, to_)


n = int(input())

print(2 ** n - 1)
hanoi(n, 1, 3)  # n개의 원판을 1번 막대에서 3번 막대로 이동
